explore_data/nbconverted/cluster_by_metadata.py

```diff
@@ -0,0 +1,277 @@
+
+
+# In[4]:
+
+
+# Select meatadata variable to color by
+metadata_variable = 'genotype'
+#metadata_variable = 'processing'
+
+
+# In[5]:
+
+
+print(expression_pao1_df.shape)
+expression_pao1_df.head()
+
+
+# In[6]:
+
+
+print(expression_pa14_df.shape)
+expression_pa14_df.head()
+
+
+# ## Parse metadata
+
+# ## Plot in PCA space
+
+# In[7]:
+
+
+# Embed expression data into low dimensional space
+pca = PCA(n_components=2)
+model = pca.fit(expression_pao1_df)
+pao1_encoded = model.transform(expression_pao1_df)
+
+pao1_encoded_df = pd.DataFrame(data=pao1_encoded,
+                               index=expression_pao1_df.index,
+                               columns=['1','2'])
+
+# Add label
+if metadata_variable == "genotype":
+    pa14_ids = list(pa14_metadata.index)
+    pao1_encoded_df['genotype'] = 'PAO1'
+    pao1_encoded_df.loc[pa14_ids,'genotype'] = 'PA14'
+elif metadata_variable == "processing":
+    pao1_ids = list(pao1_metadata.index)
+    pa14_ids = list(pa14_metadata.index)
+    pao1_encoded_df.loc[pao1_ids,"processing"] = pao1_metadata.loc[pao1_ids,"LibrarySelection"].values
+    pao1_encoded_df.loc[pa14_ids,"processing"] = pa14_metadata.loc[pa14_ids,"LibrarySelection"].values
+
+pao1_encoded_df.head()
+
+
+# In[8]:
+
+
+model = pca.fit(expression_pa14_df)
+pa14_encoded = model.transform(expression_pa14_df)
+
+pa14_encoded_df = pd.DataFrame(data=pa14_encoded,
+                               index=expression_pa14_df.index,
+                               columns=['1','2'])
+
+# Add label
+if metadata_variable == "genotype":
+    pa14_ids = list(pa14_metadata.index)
+    pa14_encoded_df['genotype'] = 'PAO1'
+    pa14_encoded_df.loc[pa14_ids,'genotype'] = 'PA14'
+elif metadata_variable == "processing":
+    pao1_ids = list(pao1_metadata.index)
+    pa14_ids = list(pa14_metadata.index)
+    pa14_encoded_df.loc[pao1_ids,"processing"] = pao1_metadata.loc[pao1_ids,"LibrarySelection"].values
+    pa14_encoded_df.loc[pa14_ids,"processing"] = pa14_metadata.loc[pa14_ids,"LibrarySelection"].values
+
+pa14_encoded_df.head()
+
+
+# In[11]:
+
+
+# Plot PAO1
+fig = pn.ggplot(pao1_encoded_df, pn.aes(x='1', y='2'))
+fig += pn.geom_point(pn.aes(color=metadata_variable), alpha=0.5)
+fig += pn.labs(x ='PC 1',
+            y = 'PC 2',
+            title = 'RNA-seq expression using PAO1 reference')
+fig += pn.theme_bw()
+fig += pn.theme(
+    legend_title_align = "center",
+    plot_background=pn.element_rect(fill='white'),
+    legend_key=pn.element_rect(fill='white', colour='white'), 
+    legend_title=pn.element_text(family='sans-serif', size=15),
+    legend_text=pn.element_text(family='sans-serif', size=12),
+    plot_title=pn.element_text(family='sans-serif', size=15),
+    axis_text=pn.element_text(family='sans-serif', size=12),
+    axis_title=pn.element_text(family='sans-serif', size=15)
+    )
+fig += pn.guides(colour=pn.guide_legend(override_aes={'alpha': 1}))
+
+print(fig)
```

Comment:
Is there a reason that this data is so colinear in PCA space? That is to say, do you have an intuition for why the PA01 points come off in spikes?

Reply:
Like the PA14 points look like I would expect something in PCA space to look, but the PA01 points feel like something is breaking

Reply:
Very good observation. I also was not expecting this shape. I'm not sure what could be going on here so something I'll need to spend some more time exploring.